src/mbgl/sprite/sprite_parser.cpp

```diff
@@ -150,6 +152,36 @@ std::optional<style::ImageContent> getContent(const JSValue& value, const char*
     return std::nullopt;
 }
 
+std::optional<style::TextFit> parseTextFit(const std::string& value) {
```

Comment:
Is this pattern used in other parts of ML/N for parsing values? Creating and filling static unordered_map, filling it with std::strings seems like overkill, especially given that there is only three constants to check against.

I'm interested to learn why this was chosen over simple string comparison?